rotkehlchen/chain/evm/decoding/hop/decoder.py

```diff
@@ -285,6 +294,133 @@ def _decode_token_swap(self, context: DecoderContext) -> DecodingOutput:
+            return DEFAULT_DECODING_OUTPUT
+
+        amount = token_normalized_value_decimals(
+            token_amount=hex_or_bytes_to_int(context.tx_log.data[160:192]),
+            token_decimals=DEFAULT_TOKEN_DECIMALS,
+        )
+        for event in context.decoded_events:
+            if (
+                event.event_type == HistoryEventType.SPEND and
+                event.event_subtype == HistoryEventSubType.NONE and
+                event.location_label == provider and
+                event.balance.amount == amount
+            ):
+                event.event_type = HistoryEventType.DEPOSIT
+                event.event_subtype = HistoryEventSubType.DEPOSIT_ASSET
+                event.counterparty = CPT_HOP
+                event.notes = f'Deposit {amount} {event.asset.symbol_or_name()} to Hop LP'
+            elif (
+                event.event_type == HistoryEventType.RECEIVE and
+                event.event_subtype == HistoryEventSubType.NONE and
+                event.location_label == provider
+            ):
+                event.event_type = HistoryEventType.RECEIVE
+                event.event_subtype = HistoryEventSubType.RECEIVE_WRAPPED
+                event.counterparty = CPT_HOP
+                event.notes = f'Receive {event.balance.amount} {event.asset.symbol_or_name()} from Hop LP'  # noqa: E501
+
+        return DEFAULT_DECODING_OUTPUT
+
+    def _decode_remove_liquidity(self, context: DecoderContext) -> DecodingOutput:
+        """Decodes a RemoveLiquidity event"""
+        if not self.base.is_tracked(provider := hex_or_bytes_to_address(context.tx_log.topics[1])):
+            return DEFAULT_DECODING_OUTPUT
+
+        if (identifier := self.swaps.get(context.tx_log.address)) is None:
+            return DEFAULT_DECODING_OUTPUT
+
+        lp_amount = token_normalized_value_decimals(
+            token_amount=hex_or_bytes_to_int(context.transaction.input_data[4:36]),
+            token_decimals=DEFAULT_TOKEN_DECIMALS,
+        )
+        token_amount = self._get_bridgeasset_amount(
+            amount_raw=hex_or_bytes_to_int(context.tx_log.data[96:128]),
+            identifier=identifier,
+        )
+        token_amount_2 = self._get_bridgeasset_amount(
+            amount_raw=hex_or_bytes_to_int(context.tx_log.data[128:160]),
+            identifier=identifier,
+        )
+        return self._handle_event_decoding(
+            decoded_events=context.decoded_events,
+            provider=provider,
+            lp_amount=lp_amount,
+            token_amounts=[token_amount, token_amount_2],
+        )
+
+    def _decode_remove_liquidity_one(self, context: DecoderContext) -> DecodingOutput:
+        """Decodes a RemoveLiquidityOne event"""
+        if not self.base.is_tracked(provider := hex_or_bytes_to_address(context.tx_log.topics[1])):
+            return DEFAULT_DECODING_OUTPUT
+
+        if (identifier := self.swaps.get(context.tx_log.address)) is None:
+            return DEFAULT_DECODING_OUTPUT
```

Comment:
what is this identifier? The identifier of the LP token? Then using `lp_token_id` is better name than identifier.

Reply:
it the identifier of asset of the saddleswap, will update to `swap_asset_id`